cmd/operator-gen-selfsigned-tls/main.go

```diff
@@ -0,0 +1,280 @@
+// Copyright 2016 The etcd-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/coreos/etcd-operator/pkg/util/k8sutil"
+	"github.com/coreos/etcd-operator/pkg/util/tlsutil"
+
+	"github.com/Sirupsen/logrus"
+)
+
+const (
+	defaultCADuration   = 10 * 365 * 24 * time.Hour
+	defaultCertDuration = 3 * 365 * 24 * time.Hour
+)
+
+var (
+	clusterName, clusterNamespace                  string
+	caSecretName, nodeSecretName, clientSecretName string
+	//Certificate Authority
+	peerCACfg, clientCACfg tlsutil.CACertConfig
+	//Node Server Certificates
+	nodePeerServerCfg, nodeClientServerCfg, etcdClientCfg tlsutil.CertConfig
+	//Node Client Certificate (for accessing etcd client interface)
+	nodeClientClientCfg tlsutil.CertConfig
+
+	peerCertDuration, clientCertDuration time.Duration
+	prettyPrint                          bool
+	outputDir                            string
+)
+
+func init() {
+	flag.StringVar(&clusterName, "cluster-name", "default-cluster", "name of etcd operator cluster to generate tls certs for")
+	flag.StringVar(&clusterNamespace, "cluster-namespace", "default", "namespace of etcd operator cluster is running in")
+
+	flag.StringVar(&caSecretName, "ca-secret-name", "etcd-operator-ca", "name of the generated self-signed CA secret")
```

Comment:
we can generate the secret name for the user based on the cluster-name, right? too many flags I feel.